packages/transport/rn-ble/src/api/transport/RNBleTransport.ts

```
@@ -144,6 +148,28 @@ export class RNBleTransport implements Transport {
     deviceId: DeviceId;
     onDisconnect: DisconnectHandler;
   }): Promise<Either<ConnectError, TransportConnectedDevice>> {
+    const existing = this._deviceConnectionsById.get(params.deviceId);
+    if (existing) {
+      const cachedDevice = this._internalDevicesById.get(params.deviceId)!;
+      return Right(
+        new TransportConnectedDevice({
+          id: params.deviceId,
+          deviceModel: cachedDevice.discoveredDevice.deviceModel,
+          type: "BLE",
+          sendApdu: (...a) => existing.sendApdu(...a),
+          transport: this.identifier,
+        }),
+      );
+    }
+
+    try {
+      await this._manager.cancelDeviceConnection(params.deviceId);
+    } catch (_) {
+      //ignore
```

[could] catch the promise instead of an empty catch block

---

catching with 
```
    await this._manager.cancelDeviceConnection(params.deviceId).catch((error) =>
      this._logger.error("[connect] cancelDeviceConnection failed", {
        data: { error },
      }),
    );
    ```